app/models/profile.rb

```diff
@@ -34,7 +21,38 @@ def activate
     end
   end
 
-  def verified?
-    verified_at.present?
+  def plain_pii
+    @_plain_pii ||= Pii::Attributes.new
+  end
+
+  def decrypt_pii(password)
+    Pii::Attributes.new_from_encrypted(encrypted_pii, password, salt)
+  end
+
+  def encrypt_pii(password, pii = plain_pii)
+    ssn = pii.ssn
+    self.ssn_signature = Pii::Fingerprinter.fingerprint(ssn) if ssn
+    self.encrypted_pii = pii.encrypted(password, salt)
+  end
+
+  def method_missing(method_sym, *arguments, &block)
+    attr_name_sym = method_sym.to_s.gsub(/=\z/, '').to_sym
+    if plain_pii.members.include?(attr_name_sym)
+      return plain_pii[attr_name_sym] if arguments.empty?
+      plain_pii[attr_name_sym] = arguments.first
```

Comment:
good question.

PII must always be encrypted before saving, and because encryption requires the user passphrase, we can't make it a `before_save` hook or equivalent. So the only attribute that really should be marked as dirty is `encrypted_pii` and marking it dirty would be technically incorrect until after `encrypt_pii` is explicitly called.

I'm starting to think that the "convenience magic" of the method_missing stuff maybe isn't worth the complexity it creates. These two do the same thing:

```ruby
profile.plain_pii[:ssn] = '1234'
profile.ssn = '1234'
```

and while I personally love the terse-ness and error checking of the 2nd form, the non-magical-ness of the 1st form might be friendlier to the code maintainer. Thoughts?